pkg/connector/group.go

```diff
@@ -406,9 +431,34 @@ func (o *groupResourceType) groupResource(ctx context.Context, group *okta.Group
 }
 
 func (o *groupResourceType) groupTrait(ctx context.Context, group *okta.Group) (*v2.GroupTrait, error) {
+	embedded := group.Embedded
+	if embedded == nil {
+		return nil, fmt.Errorf("group '%s' embedded data was nil", group.Id)
+	}
+	embeddedMap, ok := embedded.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("group '%s' embedded data was not a map", group.Id)
+	}
+	embeddedStats, ok := embeddedMap["stats"]
+	if !ok {
+		return nil, fmt.Errorf("embedded stat data was nil for group '%s'", group.Id)
+	}
+	embeddedStatsMap, ok := embeddedStats.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("group '%s' embedded stats data was not a map", group.Id)
+	}
+
+	embeddedStatsUsersCount, ok := embeddedStatsMap["usersCount"]
+	if !ok {
+		return nil, fmt.Errorf("users count not present on group '%s' embedded stats data", group.Id)
+	}
+
+	userCount := embeddedStatsUsersCount.(float64)
+
 	profile, err := structpb.NewStruct(map[string]interface{}{
-		"description": group.Profile.Description,
-		"name":        group.Profile.Name,
+		"description":        group.Profile.Description,
+		"name":               group.Profile.Name,
+		usersCountProfileKey: int64(userCount),
```

Comment:
It's interesting that we can attach metadata to the profile like this. It's also really useful information for a group in my opinion